src/CodeStyle/CSharp/CodeFixes/EditorConfigOptionsApplier.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.VisualStudio.CodingConventions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace Microsoft.CodeAnalysis.CodeStyle
+{
+    internal class EditorConfigOptionsApplier
```

Comment:
>EditorConfigOptionsApplier [](start = 19, length = 26)

This code looks extremely fragile to me. Can we avoid all the reflection magic in here if we make the following APIs public? Tagging @cyrus.najmabadi@gmail.com, as we discussed this in my previous PR as well.

1. [IEditorConfigStorageLocation](https://github.com/dotnet/roslyn/blob/dev16.0.x/src/Workspaces/Core/Portable/Options/EditorConfig/IEditorConfigStorageLocation.cs#L8)
2. [FormattingOptions.AllOptions](https://github.com/dotnet/roslyn/blob/dev16.0.x/src/Workspaces/Core/Portable/Formatting/FormattingOptions.cs#L14)
3. [CSharpFormattingOptions.AllOptions](https://github.com/dotnet/roslyn/blob/dev16.0.x/src/Workspaces/CSharp/Portable/Formatting/CSharpFormattingOptions.cs#L61)

If we still want to avoid pushing for any public API changes as part of this PR, I would **strongly** prefer that we use IVT instead of fragile reflection based code accessing internal properties/fields and invoking internal methods.

Reply:
:memo: I'll make sure we include unit tests that will fail if the reflection strategy breaks down for any reason.

Reply:
Added a test in 8a88bc9790257eed30d1270bdfca580507a48346